packages/core/data-transfer/src/strapi/remote/handlers/pull.ts

```diff
@@ -191,18 +212,55 @@ export const createPullController = handlerControllerFactory<Partial<PullHandler
       configuration: () => this.provider?.createConfigurationReadStream(),
       assets: () => {
         const assets = this.provider?.createAssetsReadStream();
+        let batch: Protocol.Client.TransferAssetFlow[] = [];
+
+        const batchLength = () => {
+          return batch.reduce(
+            (acc, chunk) => (chunk.action === 'stream' ? acc + chunk.data.byteLength : acc),
+            0
+          );
+        };
+
+        const BATCH_MAX_SIZE = 1024 * 1024; // 1MB
```

Comment:
Have you done testing with different batch sizes? Assuming message round trip is 60ms a 1gb file will still have an extra 60 seconds added.

Reply:
Yes I did, but for slow connections it would break, I'd say that in the near future we could parameterize that

Reply:
Ugh, ok.  I think ultimately we're going to need full-blown support for losing / regaining a connection to solve the problem for all cases, but this is still a good improvement.